json.go

```diff
@@ -0,0 +1,266 @@
+		}
+
+		if info.archiveSizes.rootFsDiffTarSize != 0 {
+			checkpointSizeNode["root_fs_diff_size"] = metadata.ByteToString(info.archiveSizes.rootFsDiffTarSize)
+		}
+
+		node["checkpoint_size"] = checkpointSizeNode
+
+		if stats {
+			dumpStats, err := crit.GetDumpStats(task.outputDir)
+			if err != nil {
+				return fmt.Errorf("failed to get dump statistics: %w", err)
+			}
+
+			statsNode := map[string]interface{}{
+				"freezing_time": FormatTime(dumpStats.GetFreezingTime()),
+				"frozen_time":   FormatTime(dumpStats.GetFrozenTime()),
+				"memdump_time":  FormatTime(dumpStats.GetMemdumpTime()),
+				"memwrite_time": FormatTime(dumpStats.GetMemwriteTime()),
+				"pages_scanned": dumpStats.GetPagesScanned(),
+				"pages_written": dumpStats.GetPagesWritten(),
+			}
+
+			node["criu_dump_statistics"] = statsNode
+		}
+
+		if psTree {
+			psTree, err := crit.New(nil, nil, filepath.Join(task.outputDir, "checkpoint"), false, false).ExplorePs()
+			if err != nil {
+				return fmt.Errorf("failed to get process tree: %w", err)
+			}
+
+			node["process_tree"] = buildJSONPsTree(psTree, task.outputDir)
+		}
+
+		if files {
+			fds, err := crit.New(nil, nil, filepath.Join(task.outputDir, "checkpoint"), false, false).ExploreFds()
+			if err != nil {
+				return fmt.Errorf("failed to get file descriptors: %w", err)
+			}
+
+			node["file_descriptors"] = buildJSONFds(fds)
+		}
+
+		if sockets {
+			fds, err := crit.New(nil, nil, filepath.Join(task.outputDir, "checkpoint"), false, false).ExploreSk()
+			if err != nil {
+				return fmt.Errorf("failed to get sockets: %w", err)
+			}
+
+			node["sockets"] = buildJSONSks(fds)
+		}
+
+		if mounts {
+			specDump := info.specDump
+			node["mounts"] = buildJSONMounts(specDump)
+		}
+
+		result = append(result, node)
+	}
+
+	jsonData, err := json.MarshalIndent(result, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("\nDisplaying container checkpoint JSON view:\n\n%s\n", jsonData)
+
+	return nil
+}
+
+func buildJSONMounts(specDump *spec.Spec) []map[string]interface{} {
+	var result []map[string]interface{}
+
+	for _, data := range specDump.Mounts {
+		mountNode := map[string]interface{}{
+			"destination": data.Destination,
+			"type":        data.Type,
+			"source":      data.Source,
+		}
+		result = append(result, mountNode)
+	}
+
+	return result
+}
+
+func buildJSONPsTree(psTree *crit.PsTree, checkpointOutputDir string) map[string]interface{} {
+	rootNode, err := buildJSONPsNode(psTree, checkpointOutputDir)
+	if err != nil {
+		fmt.Printf("failed to build jsonPsNode: %s", err)
+		return nil
+	}
+	return map[string]interface{}{"root": rootNode}
+}
+
+func buildJSONPsNode(psTree *crit.PsTree, checkpointOutputDir string) (map[string]interface{}, error) {
+	node := map[string]interface{}{
+		"pid":  psTree.PID,
+		"comm": psTree.Comm,
+	}
+
+	cmdline, err := getCmdline(checkpointOutputDir, psTree.PID)
```

Comment:
When adding the `cmdline` key, it may be good to first check if the `ps-tree-cmd` flag is set to true. Without this check, the current implementation will fail when using only `--ps-tree` flag since memory pages files won't be extracted in this case.